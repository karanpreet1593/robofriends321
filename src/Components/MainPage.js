import React, {Component} from 'react';
import Cardlist from '../Components/Cardlist';
import SearchBox from '../Components/SearchBox';
import CounterButton from '../Components/CounterButton';
import 'tachyons';
import './MainPage.css';
import Scroll from '../Components/Scroll';
import ErrorBoundry from '../Components/ErrorBoundry';



class MainPage extends Component {
	
    componentDidMount(){
        this.props.onRequestRobots();
    }

    filteredRobots = () => {
        const { robots, searchField } = this.props;
        return robots.filter(robot => {
          return robot.name.toLowerCase().includes(searchField.toLowerCase());
        })
      }

render(){
    const { onSearchChange,isPending} = this.props

    
    
        return isPending ?
         <h1 className='tc'  >Loading</h1> :

    (
        <div className='tc'>
          <h1 className='f1' >RoboFriends</h1>
          <CounterButton />
          < SearchBox searchChange={onSearchChange}/>
          <Scroll>
           <ErrorBoundry>
               <Cardlist robots={this.filteredRobots()} />
              </ErrorBoundry>
          </Scroll>
         </div>
    );
    }
    
}	


    


export default MainPage;