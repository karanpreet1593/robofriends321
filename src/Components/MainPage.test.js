import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './Mainpage';

let wrapper;


beforeEach(()=>{
  const mockProps ={
    onRequestRobots:jest.fn(),
    robots:[],
    searchField:'',
    isPending:false
  }
  wrapper= shallow(<MainPage {...mockProps}/>)
})

it('renders MainPage without crashing', () => {
        

  expect(wrapper).toMatchSnapshot();
});

it('it filters robots correctly', () => {
        

  expect(wrapper.instance().filteredRobots([])).toEqual([])
});

it(' filter robots correctly#2', () => {
        
  const mockProps2 ={
    onRequestRobots:jest.fn(),
    robots:[{
      id:3,
      name:'john',
      email:'john@gmail.com'
    }],
    searchField:'',
    isPending:false
  }
  const wrapper2 = shallow(<MainPage {...mockProps2}/>)
  expect(wrapper2.instance().filteredRobots()).toEqual([{
    id:3,
    name:'john',
    email:'john@gmail.com'
  }])
  
});

it(' filter robots correctly#3', () => {
        
  const mockProps3 ={
    onRequestRobots:jest.fn(),
    robots:[{
      id:3,
      name:'john',
      email:'john@gmail.com'
    }],
    searchField:'',
    isPending:false
  }
  const wrapper3 = shallow(<MainPage {...mockProps3}/>)
  expect(wrapper3.instance().filteredRobots()).toEqual([{
    id:3,
    name:'john',
    email:'john@gmail.com'
  }])
  
});