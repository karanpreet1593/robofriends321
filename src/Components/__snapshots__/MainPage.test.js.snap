// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <CounterButton />
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <Cardlist
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
